server/src/main/java/eu/fasten/server/plugins/kafka/FastenKafkaPlugin.java

```diff
@@ -168,8 +170,10 @@ public void run() {
                     handleProducing(null, System.currentTimeMillis() / 1000L, KafkaRecordKind.NORMAL);
                 }
             }
-        } catch (Exception e) {
-            logger.error("Error occurred while processing call graphs", e);
+
+        } catch (WakeupException e) {
```

Comment:
As discussed offline: the WakeupExceptions should be caught in a try catch that *only* wraps the contents of the loop and, if caught, should be swallowed. Please elaborate in the empty catch that the exception just short-cuts long-running polls.  If we move the try/catch, we do not need to handle this outside anymore, when we should actually quit, `closed` is set and ends the `while`.